crates/client/gateway/src/client/request_builder.rs

```diff
@@ -71,44 +75,74 @@ impl<'a> RequestBuilder<'a> {
         unpack(self.send_get_raw().await?).await
     }
 
-    pub async fn send_get_raw(self) -> Result<Response, SequencerError> {
-        self.client.get(self.url).headers(self.headers).query(&self.params).send().await.map_err(Into::into)
+    pub async fn send_get_raw(self) -> Result<Response<Incoming>, SequencerError> {
+        let uri = self.build_uri()?;
+
+        let mut req_builder = Request::builder().method("GET").uri(uri);
+
+        for (key, value) in self.headers.iter() {
+            req_builder = req_builder.header(key, value);
+        }
+
+        let req = req_builder.body(String::new()).unwrap();
+
+        let response: Response<Incoming> = self.client.clone().call(req).await.unwrap();
+        Ok(response)
     }
 
     #[allow(dead_code)]
     pub async fn send_post<T>(self) -> Result<T, SequencerError>
     where
         T: DeserializeOwned,
     {
-        let mut request = self.client.post(self.url);
+        let uri = self.build_uri()?;
+
+        let mut req_builder = Request::builder().method("POST").uri(uri);
```

Comment:
Consider using `http::Method::POST` instead.
```suggestion
        let mut req_builder = Request::builder().method(http::Method::POST.as_str()).uri(uri);
```